src/languages/reasonml.js

```diff
@@ -6,301 +6,63 @@ Author: Gidi Meir Morris <oss@gidi.io>
-          relevance: 0
-        }
-      ]
-    }
-  ];
-
-  const PARAMS_MODE = {
-    begin: RE_IDENT,
-    end: '(,|\\n|\\))',
-    relevance: 0,
-    contains: [
-      OPERATOR_MODE,
-      {
-        className: 'typing',
-        begin: ':',
-        end: '(,|\\n)',
-        returnBegin: true,
-        relevance: 0,
-        contains: PARAMS_CONTENTS
-      }
-    ]
-  };
-
-  const FUNCTION_BLOCK_MODE = {
-    className: 'function',
-    relevance: 0,
-    keywords: KEYWORDS,
-    variants: [
-      {
-        begin: '\\s(\\(\\.?.*?\\)|' + RE_IDENT + ')\\s*=>',
-        end: '\\s*=>',
-        returnBegin: true,
-        relevance: 0,
-        contains: [
-          {
-            className: 'params',
-            variants: [
-              { begin: RE_IDENT },
-              { begin: RE_PARAM },
-              { begin: /\(\s*\)/ }
-            ]
-          }
-        ]
-      },
-      {
-        begin: '\\s\\(\\.?[^;\\|]*\\)\\s*=>',
-        end: '\\s=>',
-        returnBegin: true,
-        relevance: 0,
-        contains: [
-          {
-            className: 'params',
-            relevance: 0,
-            variants: [ PARAMS_MODE ]
-          }
-        ]
-      },
-      { begin: '\\(\\.\\s' + RE_IDENT + '\\)\\s*=>' }
-    ]
-  };
-  MODULE_ACCESS_CONTENTS.push(FUNCTION_BLOCK_MODE);
-
-  const CONSTRUCTOR_MODE = {
-    className: 'constructor',
-    begin: RE_MODULE_IDENT + '\\(',
-    end: '\\)',
-    illegal: '\\n',
-    keywords: KEYWORDS,
-    contains: [
-      hljs.QUOTE_STRING_MODE,
-      OPERATOR_MODE,
-      {
-        className: 'params',
-        begin: '\\b' + RE_IDENT
-      }
-    ]
-  };
-
-  const PATTERN_MATCH_BLOCK_MODE = {
-    className: 'pattern-match',
-    begin: '\\|',
-    returnBegin: true,
-    keywords: KEYWORDS,
-    end: '=>',
-    relevance: 0,
-    contains: [
-      CONSTRUCTOR_MODE,
-      OPERATOR_MODE,
-      {
-        relevance: 0,
-        className: 'constructor',
-        begin: RE_MODULE_IDENT
-      }
-    ]
-  };
-
-  const MODULE_ACCESS_MODE = {
-    className: 'module-access',
-    keywords: KEYWORDS,
-    returnBegin: true,
-    variants: [
-      { begin: "\\b(" + RE_MODULE_IDENT + "\\.)+" + RE_IDENT },
-      {
-        begin: "\\b(" + RE_MODULE_IDENT + "\\.)+\\(",
-        end: "\\)",
-        returnBegin: true,
-        contains: [
-          FUNCTION_BLOCK_MODE,
-          {
-            begin: '\\(',
-            end: '\\)',
-            relevance: 0,
-            skip: true
-          }
-        ].concat(MODULE_ACCESS_CONTENTS)
-      },
-      {
-        begin: "\\b(" + RE_MODULE_IDENT + "\\.)+\\{",
-        end: /\}/
-      }
-    ],
-    contains: MODULE_ACCESS_CONTENTS
-  };
-
-  PARAMS_CONTENTS.push(MODULE_ACCESS_MODE);
-
   return {
     name: 'ReasonML',
     aliases: [ 're' ],
-    keywords: KEYWORDS,
+    keywords: {
+      $pattern: '[a-z_]\\w*!?',
+      keyword:
+        'and as asr assert begin class constraint do done downto else end esfun '
```

Comment:
Should be an array of strings, one keyword per line.  See many of the newer grammars that follow this pattern for easier maintenance.

Reply:
https://github.com/highlightjs/highlight.js/pull/3795/commits/70e83b3ee7703499bc5f98ae3493b0ec6d261577